modules/imgproc/src/thresh.cpp

```diff
@@ -1202,6 +1202,82 @@ getThreshVal_Otsu_8u( const Mat& _src )
     return max_val;
 }
 
+static double
+getThreshVal_Otsu_16u( const Mat& _src )
+{
+    Size size = _src.size();
+    if( _src.isContinuous() )
+    {
+        size.width *= size.height;
+        size.height = 1;
+    }
+
+    const int N = 65536;
+    int i, j;
+    #if CV_ENABLE_UNROLLED
+    Mat hMat(1, 4 * N, CV_32S);
+    #else
+    Mat hMat(1, N, CV_32S);
```

Comment:
You are working with garbage uninitialized data.

Use `std::vector<int>` instead or `cv::AutoBuffer`.

Reply:
It is initialized to 0 on line 1222, but switching to `cv::AutoBuffer` sounds like a good idea, especially if I'm to combine 8-bit and 16-bit Otsu